stan/math/prim/mat/meta/broadcast_array.hpp

```diff
@@ -0,0 +1,29 @@
+#include <stan/math/prim/scal/meta/broadcast_array.hpp>
+#include <Eigen/Dense>
+#include <stdexcept>
+
+#ifndef STAN_MATH_PRIM_MAT_META_BROADCAST_ARRAY_HPP
+#define STAN_MATH_PRIM_MAT_META_BROADCAST_ARRAY_HPP
+
+namespace stan {
+  namespace math {
+    namespace internal {
+      template <typename ViewElt, typename OpElt, int R, int C>
+      class empty_broadcast_array<ViewElt, Eigen::Matrix<OpElt, R, C>> {
```

Comment:
All new classes, even internal ones, should have do explaining what they do and what the parameters are.

Reply:
I thought we had agreed that we didn't need doc for internal classes?

Also, this extends `prim/scal/broadcast_array.hpp`

Reply:
Looks like we never decided: http://discourse.mc-stan.org/t/math-library-namespaces-request-for-comments/292

Also, at a first look I have no idea what this is doing (I mean it's a chunk of boilerplate but what _for_?)

Reply:
OK.  (Technically, it's specializing rather than extending---extending is subclassing, whereas specialization is a template thing.)

Reply:
OK. So is it alright if I leave this as is?